### PR TITLE
Expanded type definition for chart select to include needed options

### PR DESCRIPTION
**Pull request by krahs1**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
It allows for different types of selection option (index, field or group value)

**Related github/jira issue (required)**:
#119 

**Steps necessary to review your pull request (required)**:
 API calling setChartSelection

### CODE REVIEW

File: src/app/area/area.demo.ts [hunk 1]
```diff
@@ -13,7 +13,10 @@ export class AreaDemoComponent implements OnInit {
 
   @ViewChild(SohoLineComponent) sohoLineComponent: SohoLineComponent;
 
-  private selection: SohoLineSelected  = {groupIndex: 0, fieldName: 'name', fieldValue: 'Category B'};
+  // private selection: SohoLineSelected  = {groupIndex: 1};
```
Comment (fitzorama) [inline src/app/area/area.demo.ts]: remove all the commented lines?

File: src/soho/column/soho-column.d.ts [hunk 2]
```diff
@@ -75,13 +75,17 @@ interface SohoColumn {
...
+  // USe either index or fieldName and fieldValue;
+  fieldName?: string;
+  fieldValue?: any;
+  index?: number;
 }
 
 interface SohoColumnGroupSelected {
-  groupName: string;
-  groupValue: any;
+  // USe either index or groupName and groupValue;
```
Comment (tmcconechy) [inline src/soho/column/soho-column.d.ts]: Use ...?

Comment (krahs1) [reply src/soho/column/soho-column.d.ts]: I think its good. The code below is what shows up for me as green (new code)

File: src/app/area/area.demo.ts (hunk 1 above)
Comment (krahs1) [reply src/app/area/area.demo.ts]: I added comments to each file to indicate that the commented out code are alternate examples of how to set the selection on charts.

File: src/soho/column/soho-column.d.ts (hunk 2 above)
Comment (krahs1) [reply src/soho/column/soho-column.d.ts]: oh ok I didnt notice that

File: src/app/radar/radar.demo.ts [hunk 3]
```diff
@@ -14,7 +14,10 @@ export class RadarDemoComponent implements OnInit {
 
   @ViewChild(SohoRadarComponent) sohoRadarComponent: SohoRadarComponent;
 
-  private selection: SohoRadarSelected  = {fieldName: 'name', fieldValue: 'Samsung'};
+  // The following multiple "private selection" definitions are all examples of ways to set the selection on the chart
+  // private selection: SohoRadarSelected  = {fieldName: 'name', fieldValue: 'Samsung'};
```
[the remark this answers is not in the document]
Comment (krahs1) [reply src/app/radar/radar.demo.ts]: They are the other options for doing selection. I can remove them if you want but the are the demo files and the demo files in soho controls did the same thing. I can go either way. Let me know

File: src/app/pie/pie.demo.ts [hunk 4]
```diff
@@ -13,7 +13,10 @@ export class PieDemoComponent implements OnInit {
 
   @ViewChild(SohoPieComponent) sohoPieComponent: SohoPieComponent;
 
-  private selection: SohoPieSelected  = {fieldName: 'name', fieldValue: 'Item C'};
+  // The following multiple "private selection" definitions are all examples of ways to set the selection on the chart
+  // private selection: SohoPieSelected  = {fieldName: 'name', fieldValue: 'Item D'};
```
[the remark this answers is not in the document]
Comment (tmcconechy) [reply src/app/pie/pie.demo.ts]: Ok, Then that does make sense to keep them i think. I didnt not notice that.

File: src/app/radar/radar.demo.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (krahs1) [reply src/app/radar/radar.demo.ts]: I checked in the USe to Use change so I think thats it on changes but if I missed something let me know

**Pull request merged**